src/compiler/nir/nir_divergence_analysis.c

```diff
@@ -132,7 +132,8 @@ visit_intrinsic(bool *divergent, nir_intrinsic_instr *instr)
       break;
 
    case nir_intrinsic_reduce:
-      is_divergent = nir_intrinsic_cluster_size(instr) != 0;
+      is_divergent = nir_intrinsic_cluster_size(instr) != 0 &&
+                     nir_intrinsic_cluster_size(instr) != subgroup_size;
```

Comment:
would be better if cluster_size of subgroup_size gets set to 0 somewhere earlier in NIR

Reply:
nir_lower_subgroups() would probably be the place to do it.
shouldn't be too difficult

Reply:
Should be done in 22cbf090f81d5613ed41dee42c19b7d63fadb593 and f7209c9aa4edcf3a305861028ec5ac95fce4d6c8 .